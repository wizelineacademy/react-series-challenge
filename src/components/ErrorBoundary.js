import React, { Component } from "react";

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: props.hasError,
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true, });
    console.log("error: ", info);
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong</h1>;
    }
    return this.props.children;
  }
}

ErrorBoundary.defaultProps = {
  hasError: false
}