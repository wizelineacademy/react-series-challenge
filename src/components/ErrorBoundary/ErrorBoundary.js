import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import requests from '../../actions/requests';

export class ErrorBoundary extends Component {
	componentDidCatch(error) {
		this.props.gifsLoadError(true, error);
	}

	render() {
		if (this.props.error) {
			// You can render any custom fallback UI
			return (
				<h1>
					{this.props.errorInfo}
					<br />An error occurred
				</h1>
			);
		}
		return this.props.children;
	}
}

const mapStateToProps = ({ gifsFetch }) => {
	return {
		error: gifsFetch.error,
		errorInfo: gifsFetch.errorInfo
	};
};

const mapDispatchToProps = dispatch => {
	const { gifsLoadError } = requests.creators;
	return bindActionCreators(
		{
			gifsLoadError
		},
		dispatch
	);
};

export default connect(mapStateToProps, mapDispatchToProps)(ErrorBoundary);
