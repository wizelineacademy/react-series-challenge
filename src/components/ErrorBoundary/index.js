import React, { Component } from 'react';

class ErrorBoundary extends Component {

  constructor(props){
    super(props);
    this.state = {
      hasError: false
    }
  }

  componentDidCatch() {
    this.setState(() => ({hasError: true}));
  }

  render() {
    if (this.state.hasError){
      return <h2 stye={{color:'red'}}>{this.props.message}</h2>
    }
    return this.props.children;
  }
}

export default ErrorBoundary;