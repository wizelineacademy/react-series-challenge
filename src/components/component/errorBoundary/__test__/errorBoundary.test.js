import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from "sinon";

import ErrorBoundary from "./../index";

describe('<ErrorBoundary> ', () => {

  let wrapper;

  const ErrorComponent = ({thow}) => {
    if(thow) throw new Error('hello');
    return (<div> NOT GETTING HERE ... =( </div>);
  }


  it('Will check with the snaptshot ', () => {
    wrapper = shallow(
      <ErrorBoundary>
        <ErrorComponent />
      </ErrorBoundary>
    );
    expect(toJson(wrapper.dive())).toMatchSnapshot();
  });

  it('Will check with the snaptshot when Error happen', () => {
    wrapper = mount(
      <ErrorBoundary>
        <ErrorComponent throw />
      </ErrorBoundary>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
