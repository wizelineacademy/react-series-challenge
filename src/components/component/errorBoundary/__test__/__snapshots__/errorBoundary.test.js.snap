// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary>  Will check with the snaptshot  1`] = `
<div>
   NOT GETTING HERE ... =( 
</div>
`;

exports[`<ErrorBoundary>  Will check with the snaptshot when Error happen 1`] = `
<ErrorBoundary>
  <ErrorComponent
    throw={true}
  >
    <div>
       NOT GETTING HERE ... =( 
    </div>
  </ErrorComponent>
</ErrorBoundary>
`;
